internal/store/fleetdb.go

```diff
@@ -163,68 +164,33 @@ func (s *FleetDBAPI) AssetByID(ctx context.Context, id string) (*rtypes.Server,
 		return nil, errors.Wrap(ErrDeviceID, err.Error()+id)
 	}
 
-	asset := &rtypes.Server{ID: deviceUUID.String()}
+	asset, _, err := f.client.GetServerInventory(ctx, deviceUUID, false) // use out-of-band inventory
+	if err != nil {
+		return nil, fmt.Errorf("retrieving inventory from FleetDB: %w", err)
+	}
 
 	// query credentials
-	credential, _, err := s.client.GetCredential(ctx, deviceUUID, fleetdbapi.ServerCredentialTypeBMC)
+	credential, _, err := f.client.GetCredential(ctx, deviceUUID, fleetdbapi.ServerCredentialTypeBMC)
 	if err != nil {
-		s.registerMetric("GetCredential")
+		registerMetric("GetCredential")
 
 		return nil, errors.Wrap(ErrServerserviceQuery, "GetCredential: "+err.Error())
 	}
 
 	asset.BMCUser = credential.Username
 	asset.BMCPassword = credential.Password
 
-	// query the server object
```

Comment:
What exactly is this about?

Reply:
What you see here is conversion of FleetDB data to a `rivets.Server`: 

1. Get a server object in FleetDB "form"
2. Get its bmc address via the attributes retrieved from the server record
3. Get the server vendor/model/serial from the attributes
4. Get the components
5. Stitch it all into the `rivets.Server` structure.

The new way is:

1.  "call GetServerInventory"
2. profit

This depends on a deploy of FleetDB (cf. https://github.com/metal-toolbox/fleetdb/pull/122) but it works fine in the sandbox using FleetDB `main`.

Reply:
We don't need code to decrypt FleetDB Attributes into components if we call an API that does that for you, so I could delete the attributes code.